src/JsonSchema/Constraints/Factory.php

```diff
@@ -82,6 +87,23 @@ public function __construct(
     }
 
     /**
+     * Get package files
+     *
+     * @param string File path, relative to the root of the package
+     */
+    public function getPackageFile($path)
```

Comment:
Well, the class brings in some helpers for resolving paths. It also resolves any `$refs` encountered, and yes, stores the schemas it finds. Not sure how you would integrate one for your purpose here, but it I figure it merits a look.